libbeat/cmd/instance/beat.go

```diff
@@ -772,6 +784,47 @@ func (b *Beat) createOutput(stats outputs.Observer, cfg common.ConfigNamespace)
 	return outputs.Load(b.index, b.Info, stats, cfg.Name(), cfg.Config())
 }
 
+func (b *Beat) registerClusterUUIDFetching() error {
+	if b.Config.Output.Name() == "elasticsearch" {
+		callback, err := b.clusterUUIDFetchingCallback()
+		if err != nil {
+			return err
+		}
+		elasticsearch.RegisterConnectCallback(callback)
+	}
+	return nil
+}
+
+// Build and return a callback to fetch the Elasticsearch cluster_uuid for monitoring
+func (b *Beat) clusterUUIDFetchingCallback() (elasticsearch.ConnectCallback, error) {
+	var response struct {
+		ClusterUUID string `json:"cluster_uuid"`
+	}
+
+	callback := func(esClient *elasticsearch.Client) error {
+		status, body, err := esClient.Request("GET", "/", "", nil, nil)
+		if err != nil {
+			return errw.Wrap(err, "error querying /")
+		}
+		if status > 299 {
```

Comment:
Good question. Short answer is **yes**, redirects are handled by `esClient`.

Slightly longer answer is that `esClient` ultimately ends up calling https://golang.org/pkg/net/http/#Client.Do. By default, this will redirect up to 10 times before bailing out with a `stopped after 10 redirects` message.